pkg/signatures/public_key_verifier.go

```diff
@@ -219,7 +226,12 @@ func getVerifiedImageReference(signature oci.Signature, image *storage.Image) ([
 	var verifiedImageReferences []string
 	for _, name := range image.GetNames() {
 		if signatureImageReference == fmt.Sprintf("%s/%s",
-			name.GetRegistry(), name.GetRemote()) {
+			// Special case for the default registry we use internally for docker hub:
+			// We use docker.io, whilst the cosign / go-containerregistry library uses index.docker.io.
```

Comment:
That looks potentially like a big issue. Is this a special case on the side of `go-containerregistry` or our side? Can we be sure it's the only special case we need to handle?

Reply:
Pretty sure we (stackrox) shouldn't use `docker.io` as default, but instead should / could (?) move to `index.docker.io`.

A few references which highlight `index.docker.io == DockerHub`:

- [K8S documentation](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-by-providing-credentials-on-the-command-line)
- [Default within go-containerregistry](https://github.com/google/go-containerregistry/blob/c7270c2c2395ecae7d7ffe39d92fc0c572a33cf6/pkg/name/options.go#L18-L21)
- [Additional issue for go-containerregistry explaining it](https://github.com/google/go-containerregistry/issues/68)

Additionally, if you run `docker info`, it also specifies `index.docker.io` as registry URL:
```
...
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: **https://index.docker.io/v1/**
...
```

(Albeit, official documentation from docker is quite vague and doesn't specifically state how things should be done).

I found a better solution for it, nevertheless.

If we use `name.ParseReference`, `docker.io` will be automatically re-written to `index.docker.io`, thus matching the reference in the simple signing format. I've rewritten the code.